### PR TITLE
Feature flag for autofill site breakage reporting

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/608920331025315/1207767049586664/f 

### Description
Basic plumbing for the feature flag

### Steps to test this PR
QA-optional

### CODE REVIEW

**Comment by CDRussell** [comment]
* **#4741** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4741?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4740** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4740?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CDRussell and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/reporting/remoteconfig/UnusedAutofillSiteBreakageReportingRemoteFeatureCodegenTrigger.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+@ContributesRemoteFeature(
+    scope = AppScope::class,
+    boundType = AutofillSiteBreakageReportingFeature::class,
+    featureName = "autofillBreakageReporter",
+    settingsStore = AutofillSiteBreakageReportingRemoteSettingsPersister::class,
+    exceptionsStore = AutofillSiteBreakageReportingExceptionsPersister::class,
+)
+@Suppress("unused")
+private interface UnusedAutofillSiteBreakageReportingRemoteFeatureCodegenTrigger
```
Comment (cmonfortep) [inline autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/reporting/remoteconfig/UnusedAutofillSiteBreakageReportingRemoteFeatureCodegenTrigger.kt:30]: Why do we need this Unused interface if we can have the annotation in `AutofillSiteBreakageReportingFeature`?

File: autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/reporting/AutofillSiteBreakageReportingDataStore.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+interface AutofillSiteBreakageReportingDataStore {
+
+    suspend fun getMinimumNumberOfDaysBeforeReportPromptReshown(): Int
+    suspend fun updateMinimumNumberOfDaysBeforeReportPromptReshown(newValue: Int)
+}
+
+@SingleInstanceIn(AppScope::class)
+@ContributesBinding(AppScope::class)
+class AutofillSiteBreakageReportingDataStoreImpl @Inject constructor(
+    @ImportPasswordDesktopSync private val store: DataStore<Preferences>,
```
[the remark this answers is not in the document]
Comment (CDRussell) [reply autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/reporting/AutofillSiteBreakageReportingDataStore.kt]: You're right that this was the wrong annotation. I've added its own one which will be used for anything autofill-breakage-report related

File: autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/reporting/remoteconfig/UnusedAutofillSiteBreakageReportingRemoteFeatureCodegenTrigger.kt (hunk 1 above)
Comment (CDRussell) [reply autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/reporting/remoteconfig/UnusedAutofillSiteBreakageReportingRemoteFeatureCodegenTrigger.kt]: no good reason! removed. good suggestion!

**Review by cmonfortep — APPROVED** [review]
LGTM

**Comment by CDRussell** [comment]
### Merge activity

* **Jul 24, 5:55 PM EDT**: @CDRussell started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/4740).

**Pull request merged**